### PR TITLE
Add prometheus instrumentation to sync

### PR DESCRIPTION
**Pull request by charlesakalugwu**
This adds 5 new metrics for the sync process

- sync_info (gauge) with labels
  - plugin_version
  - image
  - period_seconds
- sync_errors_total (counter)
- sync_executions_inflight (gauge)
- sync_last_executed (gauge)
- sync_duration_seconds (summary)

/cc @jim-minter @kwoodson @Makdaam @ehashman 

```release-note
NONE
```

### CODE REVIEW

**Comment by charlesakalugwu** [comment]
e2e-create flake #1695 
/retest

File: pkg/entrypoint/sync/sync.go [hunk 1]
```diff
@@ -71,6 +120,11 @@ func start(cfg *cmdConfig) error {
 		return err
 	}
 	log.Printf("running sync for plugin %s", cs.Config.PluginVersion)
+	syncInfoGauge.With(prometheus.Labels{
+		"plugin_version": cs.Config.PluginVersion,
+		"image":          cs.Config.Images.Sync,
+		"period_seconds": fmt.Sprintf("%d", int(cfg.interval.Seconds())),
```
Comment (thekad) [inline pkg/entrypoint/sync/sync.go]: Question: is `period_seconds` always going to be different? if so, i'd strongly advise against using it as a label (as per https://prometheus.io/docs/practices/instrumentation/#do-not-overuse-labels ) and instead add another metric for it.

Comment (ehashman) [reply pkg/entrypoint/sync/sync.go]: This is an info gauge and the value is being read out of the config, so I read this as only changing if the sync pod is recreated with a new config (very infrequently).

Comment (charlesakalugwu) [reply pkg/entrypoint/sync/sync.go]: @thekad the `period_seconds` will change the least often from all labels in `syncInfoGauge`

**Pull request merged**